src/app/core/session/auth/local/local-auth.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from "@angular/core";
+import { AuthUser } from "../auth-user";
+
+@Injectable({
+  providedIn: "root",
+})
+export class LocalAuthService {
+  private static LAST_LOGGED_IN_KEY = "LAST_USER";
+
+  login(): AuthUser {
+    return this.getStoredUser(LocalAuthService.LAST_LOGGED_IN_KEY);
+  }
+
+  canLoginOffline(): boolean {
+    return !!localStorage.getItem(LocalAuthService.LAST_LOGGED_IN_KEY);
+  }
+
+  private getStoredUser(username: string): AuthUser {
+    const stored = localStorage.getItem(username);
+    if (!stored) {
+      throw Error("OFFLINE LOGIN ERROR: No user found locally");
```

Comment:
is this behavior intuitive to throw an error if the login is not possible, rather than returning false / null / ...?